xblock/core.py

```diff
@@ -90,6 +99,9 @@ def open_local_resource(cls, uri):
         unauthorized resource.
         """
 
+        if isinstance(uri, six.binary_type):
+            uri = uri.decode('utf-8')
```

Comment:
I would also prefer to be strict, but the existing implementation was flexible, so maintaining backwards compatibility requires less than optimal unicode handling.  We could issue warnings when binary objects are passed in.